File: solutions/026-remove-duplicates-from-sorted-array/solution.go
package main

import "fmt"

func removeDuplicates(nums []int) int {
	k := 0
	for i := 1; i < len(nums); i++ {
		if nums[i] != nums[k] {
			k++
			nums[k] = nums[i]
		}
	}
	return k + 1

}

func main() {
	nums := []int{0, 0, 1, 1, 1, 2, 2, 3, 3, 4}
	k := removeDuplicates(nums)
	status := "PASS"
	if k != 5 {
		status = "FAIL"
	}
	fmt.Printf("%s | k=%d nums[:k]=%v\n", status, k, nums[:k])
}

File: solutions/026-remove-duplicates-from-sorted-array/solution.py
from typing import List


class Solution:
    def remove_duplicates(self, nums: List[int]) -> int:
        k = 0
        for i in range(1, len(nums)):
            if nums[i] != nums[k]:
                k += 1
                nums[k] = nums[i]
        return k + 1


if __name__ == "__main__":
    sol = Solution()
    nums = [0, 0, 1, 1, 1, 2, 2, 3, 3, 4]
    k = sol.remove_duplicates(nums)
    status = "PASS" if k == 5 and nums[:k] == [0, 1, 2, 3, 4] else "FAIL"
    print(f"{status} | k={k}, nums[:k]={nums[:k]}")
